semgrep-core/src/core/ast/AST_generic.ml

```diff
@@ -897,10 +897,6 @@ and stmt_kind =
       tok (* 'switch', also 'select' in Go, or 'case' in Bash *)
       * condition option
       * case_and_body list (* TODO brace *)
-  (* todo: merge Match with Switch?
-   * In Scala and C# the match is infix (after the expr)
```

Comment:
maybe add a comment on Switch saying this covers also the match of OCaml/Scala/C# and also the note about that it's infix in Scala/C#
You can even add the nice comment in your PR about the translation of Match (tok, e, (pat, exp)) -> Switch ...